ratpack-micrometer-metrics/src/main/java/ratpack/micrometer/metrics/internal/DefaultBlockingExecTimingInterceptor.java

```diff
@@ -0,0 +1,44 @@
+package ratpack.micrometer.metrics.internal;
+
+import io.micrometer.core.instrument.MeterRegistry;
+import io.micrometer.core.instrument.Timer;
+import ratpack.exec.ExecInterceptor;
+import ratpack.exec.Execution;
+import ratpack.func.Block;
+import ratpack.http.Request;
+import ratpack.micrometer.metrics.BlockingExecTimingInterceptor;
+import ratpack.micrometer.metrics.MicrometerMetricsConfig;
+
+import java.util.Optional;
+
+public class DefaultBlockingExecTimingInterceptor implements BlockingExecTimingInterceptor {
+  private final MicrometerMetricsConfig config;
+  private final MeterRegistry meterRegistry;
+
+  DefaultBlockingExecTimingInterceptor(MicrometerMetricsConfig config, MeterRegistry meterRegistry) {
+    this.config = config;
+    this.meterRegistry = meterRegistry;
+  }
+
+  @Override
+  public void intercept(Execution execution, ExecInterceptor.ExecType type, Block executionSegment) throws Exception {
+    if (type == ExecInterceptor.ExecType.BLOCKING) {
+      Optional<Request> requestOpt = execution.maybeGet(Request.class);
```

Comment:
There's a comment that you added above for that says that you don't have access to the `Context`, but you should be able to just change this to `execution.maybeGet(Context.class)`. The `Context` will then contain both the `Request` and `Response`.

Reply:
Whaaaaaatt... This changes everything @johnrengelman. Removed all the duplication around setting a different tag handler for blocking exec vs. execution listeners now, so `http.server.requests` and `http.blocking.execution` will align on tags (which is better anyway). Fixed in c535c7ce8d8d973d694d0f70523a4d9ac1cd1f4f.